File: packages/sdk-node/src/jails.ts
import type { HttpClient } from "./http.js";
import type { JailRecord, JailsList, JailStatus } from "./types.js";

/** Jail-run ledger: every exec endpoint appends a record here. */
export class Jails {
  constructor(private readonly http: HttpClient) {}

  /** List most-recent jail runs, newest first. */
  async list(params?: { limit?: number; status?: JailStatus }): Promise<JailsList> {
    const query: Record<string, string | number> = {};
    if (params?.limit)  query.limit  = params.limit;
    if (params?.status) query.status = params.status;
    return this.http.request<JailsList>({
      method: "GET",
      path:   "/v1/jails",
      query,
    });
  }

  /** Fetch a single jail record (full stdout/stderr, trimmed at 16 KiB). */
  async get(id: number): Promise<JailRecord> {
    return this.http.request<JailRecord>({
      method: "GET",
      path:   `/v1/jails/${id}`,
    });
  }
}

File: packages/sdk-python/src/agentjail/jails.py
"""Jails — history of one-shot + workspace runs."""

from __future__ import annotations

from ._http import HttpClient
from .types import JailKind, JailRecord, JailStatus, JailsList


class Jails:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def list(
        self,
        *,
        limit: int | None = None,
        offset: int | None = None,
        status: JailStatus | None = None,
        kind: JailKind | None = None,
        q: str | None = None,
    ) -> JailsList:
        return self._http.request(
            "GET",
            "/v1/jails",
            params={"limit": limit, "offset": offset, "status": status, "kind": kind, "q": q},
        )

    def get(self, jail_id: int) -> JailRecord:
        return self._http.request("GET", f"/v1/jails/{jail_id}")
